frontend/packages/console-shared/src/components/drawer/Drawer.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import { DraggableData, DraggableCore } from 'react-draggable';
+import './Drawer.scss';
+
+type DrawerProps = {
+  /**
+   * Height of the drawer,
+   * should be set when used as controlled component with onChange callback
+   */
+  height?: number;
+  /**
+   * Default Value: 300
+   * default height of the drawer component,
+   * should be set when used as Uncontrolled component
+   */
+  defaultHeight?: number;
+  /**
+   * Defines the minimized state of drawer
+   * false: Height is minimum height (minimized state)
+   * true: Height is greater than minimum height
+   */
+  open?: boolean;
+  /**
+   * Maximum height drawer can be resized to.
+   */
+  maxHeight?: number | string;
+  /**
+   * Set whether the drawer is resizable or not.
+   */
+  resizable?: boolean;
+  /**
+   * Content for the Header of drawer
+   */
+  header?: React.ReactNode;
+  /**
+   * This callback is invoked while resizing the drawer.
+   * @param open boolean: false when the drawer reached minimum height (minimized state)
+   * @param height number: Height of the drawer while resizing
+   */
+  onChange?: (open: boolean, height: number) => void;
+};
+
+const useSize = <T extends HTMLElement>(): [number | undefined, (element: T) => void] => {
```

Comment:
```suggestion
const useSize = <T extends HTMLElement>(): [number, (element: T) => void] => {
```
Not sure `getBoundingClientRect` can return `undefined`.

> If all the element's border-boxes are empty, then a rectangle is returned with a width and height of zero

Ref: https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect

```
document.createElement('div').getBoundingClientRect()
// DOMRect {x: 0, y: 0, width: 0, height: 0, top: 0, …}
```